ansible_runner_service.py

```diff
@@ -156,13 +185,29 @@ def main(test_mode=False):
         app.config['WTF_CSRF_ENABLED'] = False
         return app.test_client()
 
-    # Start the API server
-    app.run(host=configuration.settings.ip_address,
-            port=configuration.settings.port,
-            threaded=True,
-            ssl_context=ssl_context,
-            debug=configuration.settings.debug,
-            use_reloader=False)
+    if configuration.settings.mode == 'prod':
+        cancel = threading.Event()
+        cancel.clear()
+
+        t = threading.Thread(
+            target=cleanup_thread,
+            args=(cancel, datetime.timedelta(days=configuration.settings.artifacts_remove_frequency).total_seconds())
```

Comment:
I don't like one thing here - the `cleanup` function "knows", that `artifacts_remove_age` is a days-based-value, but we still use `timedelta` in `main` to convert `artifacts_remove_frequency` to days. How about we pass the raw `configuration.settings.artifacts_remove_frequency` here and convert it in `cleanup_thread`?

Reply:
I would rather leave it here, we would need to call the datetime twice in the cleanup_thread, or create another unnecessary variable.

Reply:
AFAICS it could be implemented as:
```python
def remove_artifacts_thread_proc(cancel, frequency):
    frequency = datetime.timedelta(days=frequency).total_seconds()
    scheduler = sched.scheduler()
    ...
```
but this is nitpicking really - the patch looks good.